### PR TITLE
Initial commit for PostgreSql Plugin

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
1) Enable ports required by the cohesity plugin.
2) Option to install and remove the plugin from the hosts.
3) Added sample playbook.

### CODE REVIEW

File: plugins/modules/cohesity_plugin.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    script_dir = module.params.get("script_dir")
+    cmd = "rpm -ivh %s -prefix %s" % (filename, script_dir)
+    rc, stdout, stderr = module.run_command(cmd)
+    # => Any return code other than 0 is considered a failure.
+    return (True, "Successfully Installed the Cohesity plugin")
+
+
+def uninstall_plugin(module):
+    try:
+        cmd = "rpm remove cohesity-postgres-connector -y"
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_plugin.py:228]: done

File: plugins/modules/cohesity_plugin.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def main():
+    # => Load the default arguments including those specific to the Cohesity Plugin.
+    argument_spec = cohesity_common_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(choices=["present", "absent"], default="present"),
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_plugin.py:243]: We follow the same way for other agent installation as well.

File: plugins/modules/cohesity_plugin.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    platform: PostgreSQL
+"""
+
+RETURN = """
+"""
+
+class InstallError(Exception):
+    pass
+                                                                                                        
+import os
```
[the remark this answers is not in the document]
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_plugin.py:91]: ok

File: playbooks/uda-workflow/register_uda_source.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Create new Protection Source for Postgresql hosts.
+        cohesity_uda_source:
+            state: absent
+            cluster: "{{ cohesity_server }}"
+            username: "{{ cohesity_username }}"
+            password: "{{ cohesity_password }}"
+            validate_certs: "{{ cohesity_validate_certs }}"
+            source_type: PostgreSQL
+            hosts:
+             - 10.2.39.171
```
Comment (pyashish) [inline playbooks/uda-workflow/register_uda_source.yml:19]: Put a random IP in examples

File: plugins/modules/cohesity_plugin.py [hunk 5]
```diff
@@ -220,19 +210,23 @@ def download_datastore_plugin(module):
...
     try:
-        cmd = "rpm remove cohesity-postgres-connector -y"
+        cmd = "rpm remove " + COHESITY_POSTGRES_CONNECTOR + " -y"
         rc, stdout, stderr = module.run_command(cmd)
         # => Any return code other than 0 is considered a failure.
+        if rc:
+            return(False, "Failed to uninstall the Cohesity plugin")
         return (True, "Successfully Uninstalled the Cohesity plugin")
     except Exception as error:
         raise__cohesity_exception__handler(error, module)
```
Comment (pyashish) [inline plugins/modules/cohesity_plugin.py:232]: If the grep fails, then error should be , couldn't find the postgres-connector to uninstall ? be as specific as possible for multiple error exit points

File: plugins/modules/cohesity_uda_source.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+        cohesity_common_argument_spec,
+        raise__cohesity_exception__handler,
+        REQUEST_TIMEOUT,
+    )
+    from ansible_collections.cohesity.dataprotect.plugins.module_utils.cohesity_hints import (
+        get__prot_source__all, unregister_source
+    )
+    from ansible_collections.cohesity.dataprotect.plugins.modules.cohesity_source import unregister_source
+except Exception:
+    pass
```
Comment (pyashish) [inline plugins/modules/cohesity_uda_source.py:91]: ModuleNotFound exeception  ? with custom message : please install xyz module ?

**Review by pyashish — APPROVED** [review]
Minor nits, fix before merging

**Pull request merged**